### DISCUSSION TITLE
Remediating CVE-2024-21626

### DISCUSSION THREAD

**Discussion by olivierjh** [rancher/rke2#5353] [Q&A]
Snyk posted a [container breakout vulnerability](https://snyk.io/blog/leaky-vessels-docker-runc-container-breakout-vulnerabilities/) at the end of January that impacts container environments like docker + kubernetes (by extension, RKE2) and it looks like it can be [remediated by updating the local version of runc](https://security.snyk.io/vuln?search=CVE-2024-21626) to a newer version. That version of runc looks to be much newer than the version of runc [included in docker images](https://github.com/rancher/image-build-runc/blob/master/Dockerfile#L18) for [ RKE2's stable release channel](https://update.rke2.io/v1-release/channels) (v1.26.12+rke2r1).

What does Rancher Labs recommend for remediating this CVE until a patch hardening the vulnerbility is developed and released?

If there's a better place to discuss this issue, please let me know.

**Comment by olivierjh** [comment]
From what I understand, you can't upgrade the version of runc or containerd underlying RKE2 without upgrading RKE2 itself. I'm very interested in the ETA on a patched version of RKE2 if one doesn't already exist, but I'm scanning the available versions as we speak.

**Comment by olivierjh** [comment]
Looks like there's an open RKE2 issue to patch the vulnerability: https://github.com/rancher/rke2/issues/5340

**Reply by brandond** [reply]
Yes. A new release of RKE2 that includes runc v1.1.12 will be available soon. 

With regards to image-build-runc, the actual tag is taken from the build tag. You're just looking at the default. Ref: https://github.com/rancher/image-build-runc/releases/tag/v1.1.12-build20240201

**Reply by megabreit** [reply]
What about older RKE2 versions? Like 1.25? Or earlier? E.g. stable Harvester versions run on 1.24 and 1.25...

**Reply by brandond** [reply]
Those releases are all end of life. We do not continue to put out new patch releases on minors once they are end of life upstream.

**Reply by megabreit** [reply]
I understand... so I'm curious how your Harvester colleagues will deal with it. Harvester is considered an appliance and manual upgrades of any component are not supported.

**Reply by brandond** [reply]
I would probably ask on the harvester project.

**Reply by markhillgit** [reply]
For Harvester the updates to rke2 1.26 and 1.27 containing these fixes will be incorporated into Harvester 1.2.2 and 1.3.0 asap.

**Reply by markhillgit** [reply]
People on 1.1.x will have to go to 1.2.1 and then 1.2.2 because we cannot make the leap from 1.24 directly, but Harvester 1.2.x is stable.

**Reply by megabreit** [reply]
I did. The answer was unexpected. But I think you guys need to discuss this internally :-)

**Reply by brandond** [reply]
The answer provided above? We don't do LTSS for Kubernetes releases, the policy has always been that you need to be on a supported minor to receive patches. That is not different just because the cluster in question is embedded in Harvester.

**Reply by megabreit** [reply]
I understand your policy. But maybe you can put yourself in my shoes for a moment: I have a support contract. And I have a vulnerability that needs to be fixed asap. In this case I don't want to get a finger pointing answer from one department of the same company to the other. Harvester is a "sort of" LTSS product, as long as RKE2 is involved, basically because of their slow release policy. So you (as in Suse) have created an LTSS product without being able to support it. This is the thing you (again Suse) need to sort out. Customers expect it.
